### PR TITLE
Prometheus: Add interpolation of groupByKeys query parameters into expr

### PR DESCRIPTION
**Pull request by kylebrandt**
**What is this feature?**
Takes a `groupByKeys` parameter `[]string` in the query and applies it to aggregate expressions.

For example:  `sum(http_requests_total)` becomes `sum by (job) (http_requests_total)`  if the groupyBy param is `["job"]`

**Special notes for your reviewer:**

Also under the promQLScope flag for now, also needs `groupByVariable` flag.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/promlib/models/query.types.json [hunk 1]
```diff
@@ -69,6 +69,13 @@
               "type": "string",
               "x-enum-description": {}
             },
+            "groupByKeys": {
+              "description": "Group By parameters to apply to aggregate expressions in the query",
```
Comment (bergquist) [inline pkg/promlib/models/query.types.json:73]: This is used to generate schemas. right? Perhaps we should make an comment about it being experimental.

File: pkg/promlib/models/scope_test.go [hunk 2]
```diff
@@ -92,7 +92,105 @@ func TestApplyQueryFilters(t *testing.T) {
...
+		query     string
+		groupBy   []string
+		expected  string
+		expectErr bool
+	}{
+		{
+			name:      "GroupBy with no aggregate expression",
+			groupBy:   []string{"job"},
+			query:     `http_requests_total`,
+			expected:  `http_requests_total`,
```
Comment (bergquist) [inline pkg/promlib/models/scope_test.go:119]: I think the expected result here would be `sum by (job) (http_requests_total)`. But up for discussion and we can also consider queries without sum functions a next step.

File: pkg/promlib/models/scope_test.go [hunk 3]
```diff
@@ -92,7 +92,105 @@ func TestApplyQueryFilters(t *testing.T) {
...
+			name:  "Adhoc filters with more complex expression",
+			query: `sum(capacity_bytes{job="prometheus"} + available_bytes{job="grafana"}) / 1024`,
+			adhocFilters: []ScopeFilter{
+				{Key: "job", Value: "alloy", Operator: FilterOperatorEquals},
+			},
+			scopeFilters: []ScopeFilter{
+				{Key: "vol", Value: "/", Operator: FilterOperatorEquals},
+			},
+			groupby:   []string{"job"},
+			expected:  `sum by (job) (capacity_bytes{job="alloy",vol="/"} + available_bytes{job="alloy",vol="/"}) / 1024`,
```
Comment (bergquist) [inline pkg/promlib/models/scope_test.go:186]: 👍

**Review by bergquist — APPROVED** [review]
I think we can merge this but I'd love to dig into if queries without sum functions should be wrapped in sum or not later on. Boths options would be valid answers so we might have to look at examples for figure out which one makes the most sense.

**Comment by kylebrandt** [comment]
> queries without sum functions should be wrapped in sum or not later on

Was thinking this as well. But also wanted to see what examples we come up with. My guess is that if a dashboard is showing some sort of side-by-side view, where one side is aggregated and the other side isn't -- then we would want it this way. 

That being said I think it is more likely we might want to add an aggregation when there isn't one, but that is a more complex modification of queries as well - so seemed logical to look into this separately.

File: pkg/promlib/models/query.types.json (hunk 1 above)
Comment (kylebrandt) [reply pkg/promlib/models/query.types.json:73]: I'm not sure what this json is used for, but experimental is also true of AdhocFilters and Scopes props ... so going to not worry about this right now :-)

**Pull request merged**